cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -708,239 +830,275 @@ func TestStartDeletion(t *testing.T) {
-						return false, nil, nil
-					}
-					eviction := createAction.GetObject().(*policyv1beta1.Eviction)
-					if eviction == nil {
-						return false, nil, nil
-					}
-					if tc.failedPodDrain[eviction.Name] {
-						return true, nil, fmt.Errorf("SIMULATED ERROR: won't evict")
+				fakeClient.Fake.AddReactor("get", "pods",
+					func(action core.Action) (bool, runtime.Object, error) {
+						return true, nil, errors.NewNotFound(apiv1.Resource("pod"), "whatever")
+					})
+				// Hook node update to gather all taint updates, and to fail the update for certain nodes to simulate errors.
+				fakeClient.Fake.AddReactor("update", "nodes",
+					func(action core.Action) (bool, runtime.Object, error) {
+						nodesLock.Lock()
+						defer nodesLock.Unlock()
+						update := action.(core.UpdateAction)
+						obj := update.GetObject().(*apiv1.Node)
+						if tc.failedNodeTaint[obj.Name] {
+							return true, nil, fmt.Errorf("SIMULATED ERROR: won't taint")
+						}
+						nt := nodeTaints{
+							nodeName: obj.Name,
+						}
+						for _, taint := range obj.Spec.Taints {
+							nt.taints = append(nt.taints, taint)
+						}
+						taintUpdates <- nt
+						nodesByName[obj.Name] = obj.DeepCopy()
+						return true, obj, nil
+					})
+				// Hook eviction creation to gather which pods were evicted, and to fail the eviction for certain pods to simulate errors.
+				fakeClient.Fake.AddReactor("create", "pods",
+					func(action core.Action) (bool, runtime.Object, error) {
+						createAction := action.(core.CreateAction)
+						if createAction == nil {
+							return false, nil, nil
+						}
+						eviction := createAction.GetObject().(*policyv1beta1.Eviction)
+						if eviction == nil {
+							return false, nil, nil
+						}
+						if tc.failedPodDrain[eviction.Name] {
+							return true, nil, fmt.Errorf("SIMULATED ERROR: won't evict")
+						}
+						deletedPods <- eviction.Name
+						return true, nil, nil
+					})
+
+				// Hook node deletion at the level of cloud provider, to gather which nodes were deleted, and to fail the deletion for
+				// certain nodes to simulate errors.
+				provider := testprovider.NewTestCloudProvider(nil, func(nodeGroup string, node string) error {
+					if tc.failedNodeDeletion[node] {
+						return fmt.Errorf("SIMULATED ERROR: won't remove node")
 					}
-					deletedPods <- eviction.Name
-					return true, nil, nil
+					deletedNodes <- node
+					return nil
 				})
-
-			// Hook node deletion at the level of cloud provider, to gather which nodes were deleted, and to fail the deletion for
-			// certain nodes to simulate errors.
-			provider := testprovider.NewTestCloudProvider(nil, func(nodeGroup string, node string) error {
-				if tc.failedNodeDeletion[node] {
-					return fmt.Errorf("SIMULATED ERROR: won't remove node")
+				testSet.nodeGroup.SetCloudProvider(provider)
+				provider.InsertNodeGroup(testSet.nodeGroup)
+				for _, node := range nodesByName {
+					provider.AddNode(testSet.nodeGroup.Id(), node)
```

Comment:
I have changed line 956 to 959 to use nodegroup from `testSet` instead of hardcoding `testNg1`